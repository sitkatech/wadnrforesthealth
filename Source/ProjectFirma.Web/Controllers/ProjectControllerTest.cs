using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using ProjectFirma.Web.UnitTestCommon;

namespace ProjectFirma.Web.Controllers
{
    [TestFixture]
    public class ProjectControllerTest : DatabaseDirectAccessTestFixtureBase
    {
        /// <summary>
        ///  **** If this unit test fails, that means a new FK to dbo.Project was added.
        /// Make sure this FK/table is added to the delete statements in the stored procedure 'dbo.pBulkDeleteProjects'
        /// </summary>
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void FlagChangesToForeignKeysReferencingProjectTableForDevelopersToUpdateBulkDeleteProjectsStoredProcedure()
        {
            const string sqlQueryString = @"
                        SELECT  
                            tab1.name AS [table],
                            col1.name AS [column],
                            obj.name AS FK_NAME
                        FROM sys.foreign_key_columns fkc
                        INNER JOIN sys.objects obj
                            ON obj.object_id = fkc.constraint_object_id
                        INNER JOIN sys.tables tab1
                            ON tab1.object_id = fkc.parent_object_id
                        INNER JOIN sys.columns col1
                            ON col1.column_id = parent_column_id AND col1.object_id = tab1.object_id
						INNER JOIN sys.tables tab2
							ON tab2.object_id = fkc.referenced_object_id
						INNER JOIN sys.schemas sch
							ON tab1.schema_id = sch.schema_id
						WHERE
							tab2.name = 'Project' and sch.name = 'dbo'
                        ORDER BY tab1.name;";
            var result = ExecAdHocSql(sqlQueryString);
            Approvals.Verify(result.TableToHumanReadableString());
        }

        /// <summary>
        ///  **** If this unit test fails, that means a new FK to dbo.ProjectUpdateBatch was added.
        /// Make sure this FK/table is added to the delete statements in the stored procedure 'dbo.pBulkDeleteProjects'
        /// </summary>
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void FlagChangesToForeignKeysReferencingProjectUpdateBatchTableForDevelopersToUpdateBulkDeleteProjectsStoredProcedure()
        {
            const string sqlQueryString = @"
                        SELECT  
                            tab1.name AS [table],
                            col1.name AS [column],
                            obj.name AS FK_NAME
                        FROM sys.foreign_key_columns fkc
                        INNER JOIN sys.objects obj
                            ON obj.object_id = fkc.constraint_object_id
                        INNER JOIN sys.tables tab1
                            ON tab1.object_id = fkc.parent_object_id
                        INNER JOIN sys.columns col1
                            ON col1.column_id = parent_column_id AND col1.object_id = tab1.object_id
						INNER JOIN sys.tables tab2
							ON tab2.object_id = fkc.referenced_object_id
						INNER JOIN sys.schemas sch
							ON tab1.schema_id = sch.schema_id
						WHERE
							tab2.name = 'ProjectUpdateBatch' and sch.name = 'dbo'
                        ORDER BY tab1.name;";
            var result = ExecAdHocSql(sqlQueryString);
            Approvals.Verify(result.TableToHumanReadableString());
        }

        [TestFixtureSetUp]
        public void TestFixtureSetup()
        {
            BaseFixtureSetup();
        }

        [TestFixtureTearDown]
        public void TestFixtureTeardown()
        {
            BaseFixtureTeardown();
        }

    }
}
