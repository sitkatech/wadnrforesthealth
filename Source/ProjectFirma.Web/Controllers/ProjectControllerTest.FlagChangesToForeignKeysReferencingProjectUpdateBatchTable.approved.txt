table	column	FK_NAME
PerformanceMeasureActualUpdate	ProjectUpdateBatchID	FK_PerformanceMeasureActualUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectCountyUpdate	ProjectUpdateBatchID	FK_ProjectCountyUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectCustomAttributeUpdate	ProjectUpdateBatchID	FK_ProjectCustomAttributeUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectDocumentUpdate	ProjectUpdateBatchID	FK_ProjectDocumentUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectExemptReportingYearUpdate	ProjectUpdateBatchID	FK_ProjectExemptReportingYearUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectExternalLinkUpdate	ProjectUpdateBatchID	FK_ProjectExternalLinkUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectFundingSourceUpdate	ProjectUpdateBatchID	FK_ProjectFundingSourceUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectGrantAllocationExpenditureUpdate	ProjectUpdateBatchID	FK_ProjectGrantAllocationExpenditureUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectGrantAllocationRequestUpdate	ProjectUpdateBatchID	FK_ProjectGrantAllocationRequestUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectImageUpdate	ProjectUpdateBatchID	FK_ProjectImageUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectLocationStagingUpdate	ProjectUpdateBatchID	FK_ProjectLocationStagingUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectLocationUpdate	ProjectUpdateBatchID	FK_ProjectLocationUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectNoteUpdate	ProjectUpdateBatchID	FK_ProjectNoteUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectOrganizationUpdate	ProjectUpdateBatchID	FK_ProjectOrganizationUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectPersonUpdate	ProjectUpdateBatchID	FK_ProjectPersonUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectPriorityLandscapeUpdate	ProjectUpdateBatchID	FK_ProjectPriorityLandscapeUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectRegionUpdate	ProjectUpdateBatchID	FK_ProjectRegionUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectUpdate	ProjectUpdateBatchID	FK_ProjectUpdate_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectUpdateHistory	ProjectUpdateBatchID	FK_ProjectUpdateHistory_ProjectUpdateBatch_ProjectUpdateBatchID
ProjectUpdateProgram	ProjectUpdateBatchID	FK_ProjectUpdateProgram_ProjectUpdateBatch_ProjectUpdateBatchID
TreatmentUpdate	ProjectUpdateBatchID	FK_TreatmentUpdate_ProjectUpdateBatch_ProjectUpdateBatchID